lsp-methods.el

```diff
@@ -1421,9 +1421,25 @@ If title is nil, return the name for the command handler."
   "Ask the server to format this document."
   (interactive)
   (let ((edits (lsp--send-request (lsp--make-request
-                                    "textDocument/formatting"
-                                    (lsp--make-document-formatting-params)))))
-    (save-excursion (lsp--apply-text-edits edits))))
+                                   "textDocument/formatting"
+                                   (lsp--make-document-formatting-params)))))
+    (if (fboundp 'replace-buffer-contents)
+        (let ((content (buffer-string))
+              (buffer (get-buffer-create " *lsp-formatting*")))
```

Comment:
Why not use `generate-new-buffer` to get a unique buffer name?

Reply:
Or just `with-temp-buffer`, then you can also remove the `unwind-protect`.

Reply:
@phst `with-temp-buffer` cannot be used. It would be deleted in the `replace-buffer-contents` call